HelpSource/Guides/MultiClient_Setups.schelp

```diff
@@ -0,0 +1,176 @@
+
+OSC communication between SC and its sound server offers many options for network music: Multiple computers can run both supercollider and associated sound servers.
+For clarity, the "server program" refers a running scsynth or supernova process. The "server object" AKA "client" is the server's representation in sclang, such as code::Server.local, s, Server(\elsewhere, NetAddr("123.234.56.78"))::.
+
+subsection:: What are clientIDs for, and how do servers get them?
+
+When more than one user plays on a given server,
+some resources need to be shared between users/clients:
+list::
+## permanent and temporary nodeIDs (handled by s.nodeAllocator),
+## private control and audio bus channels (handled by s.audioBusAllocator, s.controlBusAllocator)
+## buffer numbers (handled by s.bufferAllocator).
+::
+
+In the simple default case, there is only a single user/client, who always gets clientID 0, and all the available resources (i.e. the full ranges of allocator numbers).
+
+When multiple clients log in, this is what happens:
+list::
+## On startup, scsynth (the server program) gets a fixed limit of maxLogins.
+## When a new local or remote server object/client registers on this scsynth process,
+scsynth sends back the next free clientID.
+## Exception: If the server object was given a clientID on creation, it keeps that clientID, to allow fixed clientID distribution. (This variant may become deprecated in the future.)
+::
+
+subsection:: Code examples and tests
+
+A. Test the clientID method by hand
+
+Note:: Setting clientID_ by hand is not recommended, as it can be confusing.
+Here it is only done for testing the method!::
+
+code::
+
+s.options.maxLogins = 12;
+// these work and post info
+s.clientID_(1);
+s.clientID_(2);
+s.clientID_(11);
+// guards against bad input - warns and keeps clientID
+s.clientID_(nil);
+s.clientID_("abc");
+s.clientID_(16);
+s.clientID_(-234);
+// reset to default:
+s.clientID_(0);
+
+
+// Recommended usage for fixed clientIDs - set clientID on creation.
+
+s.options.maxLogins = 12;
+
+z = Server(
+	\remote4,
+	// NetAddr("168.192.1.2", 57110), // netAddr of the other computer in my network
+	NetAddr("127.0.0.1", 57105), // for testing, on this machine, different port
+	s.options,		// make sure all remote servers have the same options
+	4				// and if desired, set fixed client by some scheme
+);
+
+// now s knows it can change clientID from server login response
+// (because userSpecifiedClientID is false)
+s.userSpecifiedClientID;
+// and z knows to keep its clientID
+z.userSpecifiedClientID;
+
+::
+
+subsection:: Separate defaultGroups and easy-to trace nodeIDs
+
+Every client registering with a server has its own defaultGroup. All nodes belonging to one client are in its defaultGroup and can be specifically addressed, so e.g. freeAll can release only one's own nodes, and not those of other clients on this server.
+For details, see the ReadableNodeIDAllocator class.
+
+code::
```

Comment:
I'd like to keep it here for the overview, and extend it further in a ReadableNodeIDAllocator helpfile to be written.